zcash_primitives/src/sapling/note_encryption.rs

```diff
@@ -1450,22 +1452,16 @@ mod tests {
             })
             .collect();
 
-        let res = batch::try_note_decryption(&[invalid_ivk.clone(), valid_ivk.clone()], &outputs);
-        assert_eq!(res.len(), 20);
-        // The batched trial decryptions with invalid_ivk failed.
-        assert_eq!(&res[..10], &vec![None; 10][..]);
-        for (result, (_, output)) in res[10..].iter().zip(outputs.iter()) {
-            // Confirm that the outputs should indeed have failed with invalid_ivk
-            assert_eq!(
-                try_sapling_note_decryption(&TEST_NETWORK, height, &invalid_ivk, output),
-                None
-            );
-
```

Comment:
This check should have been left in.

Reply:
I guess that the check can now be that it fully fails if you only pass in an invalid Ivk.